chasm/lib/activity/activity.go

```diff
@@ -259,6 +260,25 @@ func (a *Activity) HandleFailed(ctx chasm.MutableContext, req *historyservice.Re
 	return &historyservice.RespondActivityTaskFailedResponse{}, nil
 }
 
+// HandleCanceled updates the activity on activity canceled.
+func (a *Activity) HandleCanceled(ctx chasm.MutableContext, request *historyservice.RespondActivityTaskCanceledRequest) (
+	*historyservice.RespondActivityTaskCanceledResponse, error) {
```

Comment:
nit: this isn't the formatting style that the rest of the codebase uses. They do
```go
func (a *Activity) HandleCanceled(
	ctx chasm.MutableContext,
	request *historyservice.RespondActivityTaskCanceledRequest,
) (*historyservice.RespondActivityTaskCanceledResponse, error) {
```